pkgs/by-name/di/dita-ot/package.nix

```diff
@@ -0,0 +1,62 @@
+{ stdenv
+, fetchzip
+, bash
+, jdk17
+, nix
+, coreutils
+, lib
+, makeWrapper
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  name = "dita-ot";
+  version = "4.2.1";
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+  buildInputs = [
+    bash
+    jdk17
+  ];
+
+  src = fetchzip {
+    url = "https://github.com/dita-ot/dita-ot/releases/download/${finalAttrs.version}/dita-ot-${finalAttrs.version}.zip";
+    hash = "sha256-PtscKsQwW1uBu0OfVHi7/d9JujGx12nZT+W3uMPQQWc=";
+  };
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin $out/$name/
+    cp -r $src/* $out/$name/
+    chmod -R 777 $out/$name/
+
+    wrapProgram "$out/$name/bin/dita" \
+      --prefix PATH : "${lib.makeBinPath [ jdk17 ]}" \
+      --set-default JDK_HOME "$jdk17" \
+      --set-default JAVA_HOME "$jdk17" \
+      --set-default ANDROID_HOME "$jdk17"
+
+    ln -sf "$out/$name/bin/dita" "$out/bin/dita"
+
+    runHook postInstall
+  '';
+
+  doInstallCheck = true;
+  installCheckPhase = ''
+    runHook preInstallCheck
+    $out/bin/dita --version
+    runHook postInstallCheck
+  '';
```

Comment:
Perhaps replace this [with `passthru.tests.version`](https://nixos.org/manual/nixpkgs/stable/#tester-testVersion), as it’s [preferred](https://nixos.org/manual/nixpkgs/stable/#var-meta-tests).

Reply:
Very good idea! It looks much leaner now.